xtext-website/documentation/103_domainmodelnextsteps.md

```diff
@@ -235,46 +276,34 @@ This tutorial is about testing the parser and the linker for the *Domainmodel*.
             assertSame(entity, entity.features.head.type)
         }
     ```
+2. In addition, the utility class [ValidationTestHelper]({{site.src.xtext_core}}/org.eclipse.xtext.testing/src/org/eclipse/xtext/testing/validation/ValidationTestHelper.java) allows to test the custom validation rules written for the language. Both valid and invalid models can be tested.
 
-1.  After saving the Xtend file, it is time to run the test. Select *Run As &rarr; JUnit Test* from the editor's context menu.
-
-## Creating Custom Validation Rules {#tutorial-validation}
-
-One of the main advantages of DSLs is the possibility to statically validate domain-specific constraints. Because this is a common use case, Xtext provides a dedicated hook for this kind of validation rules. In this lesson, we want to ensure that the name of an *Entity* starts with an upper-case letter and that all features have distinct names across the inheritance relationship of an *Entity*.
-
-Locate the class *DomainmodelValidator* in the package *org.example.domainmodel.validation* of the language project. Defining the constraint itself is only a matter of a few lines of code:
-
-```java
-@Check
-public void checkNameStartsWithCapital(Entity entity) {
-    if (!Character.isUpperCase(entity.getName().charAt(0))) {
-        warning("Name should start with a capital", 
-            DomainmodelPackage.Literals.TYPE__NAME);
-    }
-}
-```
-
-Any name for the method will do. The important thing is the [Check]({{site.src.xtext_core}}/org.eclipse.xtext/src/org/eclipse/xtext/validation/Check.java) annotation that advises the framework to use the method as a validation rule. If the name starts with a lower case letter, a warning will be attached to the name of the *Entity*.
-
-The second validation rule is straight-forward, too. We traverse the inheritance hierarchy of the *Entity* and look for features with equal names.
+    ```xtend
+        
+ 	    @Inject extension ValidationTestHelper
+
+	    @Test
+	    def testNameStartsWithCapitalWarning() {
+		parseHelper.parse(
```

Comment:
```
val entity = parseHelper.parse(...)
entity.assertWarning(...)
```
Please try to make it simple. If possible, please avoid using constructs like "extension" methods as it might confuse a new user.  If you want to use it, please use it in the second method and provide pointers and a bit more explanation about the syntax so that the user knows what is happening here.

Reply:
I'd like to use it, and, as a user myself, I'd find more documentation and examples better. Now I've written the tests firstly without the extensions, I've added a pointer to the Xtend documentation on the extension and provided a sample implementation with them.